### DISCUSSION TITLE
Performance of POS function

### DISCUSSION THREAD

**Discussion by Peter-Jacob** [adesutherland/CREXX#101] [General]
I was concerned about POS' performance, it consists 40 lines of asm statements and contains 2 nested loops. 
I ran a performance test:
- create a long string of 500 '1's and added a '9' at the end
- did a POS on the '9', actually, it has to run through the entire string to find it in the last position. 
- ran it in a 100,000 loop
I think the performance is amazing: it took 3 seconds! For a reduced string length of 100 chars it's one second.

I think this clearly indicates our rxas stuff is in most cases fast enough. 

Update: I ran this rexx in BREXX:
```
x=copies('1',500)'9'
say time('l')
do i=1 to 100000
   y=pos('9',x)
end
say y i
say time('l')
```
```
12:52:13.181296
501 100001
12:52:13.268124
```
conclusion: rxas is far too slow, very disappointing!

(My CPU is an AMD Ryzen 5 PRO 4650G Radeon Graphics 3.70 GHz)

**Comment by rvjansen** [comment]
That is a great result. It means we will fallback to native code on only a very small number of occasions. I am curious as to how the comparison to the rxc compiled versions will be; that will also give us (well, Adrian for the moment) direction how to tune that.

René.
